library/Icinga/Util/Csp.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Icinga\Util;
+
+use Icinga\Web\Response;
+use Icinga\Web\Session;
+use Icinga\Web\Window;
+
+class Csp
+{
+    /** @var self */
+    private static $instance = null;
```

Comment:
This targets all `self` references which in fact prevents that this class is ever subclassed. Please use `static` everywhere instead. `self` as return type declaration is fine though.